include/tiramisu/core.h

```diff
@@ -1144,13 +1196,21 @@ class buffer
       * the common case), the function that was created automatically
       * during Tiramisu initialization will be used (we call that
       * function the "implicit function").
+      * 
+      * \p corr is the name of the cpu buffer corresponding to a gpu buffer. 
+      * This field is only set, when we creat a gpu buffer.
+      *
+      * \p location specifies where to store the buffer. Either in host or 
+      *  one of the GPU memories.
       *
       * Buffer names should not start with _ (an underscore).
       * Names starting with _ are reserved names.
       */
     buffer(std::string name, std::vector<tiramisu::expr> dim_sizes,
            tiramisu::primitive_t type, tiramisu::argument_t argt,
-	   tiramisu::function *fct = global::get_implicit_function());
+           tiramisu::function *fct = global::get_implicit_function(), 
+	   std::string corr = "", cuda_ast::memory_location location = cuda_ast::memory_location::host);
```

Comment:
I just noticed that you are modifying the buffer() constructor.  We better avoid this. I suggest that you create a new constructor that call the old constructor and set the location.

Reply:
For the documentation of the new constructor just say that it is similar to the original constructor and document the new arguments.

Reply:
I have just left corr , because else the user will have to write for each gpu buffer a line of code, and this will.
and i couldn't use an other constructor, in both cases with default values of some fields or not. that why i have kept only corr. the other we will use the Tiramisu API like u said